applications/GeoMechanicsApplication/custom_utilities/check_utilities.h

```diff
@@ -39,4 +40,123 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) CheckUtilities
+    {
+        CheckAvailabilityOnly(rVariable);
+        CheckRangeBounds(rVariable, mDefaultLowerBound, std::nullopt);
+    }
+
+    template <typename T>
+    void Check(const Variable<T>& rVariable, double UpperBound) const
+    {
+        CheckAvailabilityOnly(rVariable);
+        CheckRangeBounds(rVariable, mDefaultLowerBound, UpperBound);
+    }
+
+    template <typename T>
+    void Check(const Variable<T>& rVariable, double LowerBound, double UpperBound) const
+    {
+        CheckAvailabilityOnly(rVariable);
+        CheckRangeBounds(rVariable, LowerBound, UpperBound);
+    }
+
+    template <typename T>
+    void CheckAvailabilityOnly(const Variable<T>& rVariable) const
+    {
+        if (!mrProperties.Has(rVariable))
+            KRATOS_ERROR << rVariable.Name() << " does not exist in the " << mrPrintName
+                         << " with Id " << mId << "." << std::endl;
+    }
+
+    template <typename T>
+    void CheckAvailabilityAndEmpty(const Variable<T>& rVariable) const
+    {
+        CheckAvailabilityOnly(rVariable);
+        if (mrProperties[rVariable].empty())
+            KRATOS_ERROR << rVariable.Name() << " is empty in the " << mrPrintName << " with Id "
+                         << mId << "." << std::endl;
+    }
+
+    void CheckPermeabilityProperties(size_t Dimension) const;
+
+private:
+    const Properties& mrProperties;
+    const std::string mrPrintName;
+    const std::size_t mId;
+    mutable Bounds    mRangeBoundsType;
+    const double      mDefaultLowerBound = 0.0;
+
+    template <typename T>
+    void CheckRangeBounds(const Variable<T>& rVariable, double LowerBound, std::optional<double> UpperBound) const
+    {
+        const auto value = mrProperties[rVariable];
+        bool       in_range;
+        using enum CheckProperties::Bounds;
+
+        switch (mRangeBoundsType) {
+        case AllExclusive:
+            in_range = (value > LowerBound && (!UpperBound || value < *UpperBound));
+            break;
+        case AllInclusive:
+            in_range = (value >= LowerBound && (!UpperBound || value <= *UpperBound));
+            break;
+        case InclusiveLowerAndExclusiveUpper:
+            in_range = (value >= LowerBound && (!UpperBound || value < *UpperBound));
+            break;
+        case ExclusiveLowerAndInclusiveUpper:
+            in_range = (value > LowerBound && (!UpperBound || value <= *UpperBound));
+            break;
+        default:
+            KRATOS_ERROR << " Unknown type of range bounds";
+            break;
+        }
+        if (!in_range) {
+            std::ostringstream print_range;
+            const auto         include_lower_bound = (mRangeBoundsType == AllInclusive) ||
+                                             (mRangeBoundsType == InclusiveLowerAndExclusiveUpper);
+            const auto include_upper_bound = (mRangeBoundsType == AllInclusive) ||
+                                             (mRangeBoundsType == ExclusiveLowerAndInclusiveUpper);
```

Comment:
I think if the upper bound doesn't exist, then we should also not include the upper bound (so we don't end up with e.g. `(0.0, -]`)
```suggestion
            const auto include_upper_bound = UpperBound && ((mRangeBoundsType == AllInclusive) ||
                                             (mRangeBoundsType == ExclusiveLowerAndInclusiveUpper));
```